server/monitoring/azureAppInsights.ts

```diff
@@ -27,9 +30,23 @@ export function initialiseAppInsights(): void {
 
 export function buildAppInsightsClient(name = defaultName()): TelemetryClient {
   if (process.env.APPINSIGHTS_INSTRUMENTATIONKEY) {
-    defaultClient.context.tags['ai.cloud.role'] = name
-    defaultClient.context.tags['ai.application.ver'] = version()
+    if (defaultClient) {
+      defaultClient.context.tags['ai.cloud.role'] = name
+      defaultClient.context.tags['ai.application.ver'] = version()
+    }
     return defaultClient
   }
   return null
 }
+
+export const trackEvent = (eventName: string, req: Request) => {
+  const requestBody = req.body
+
+  if (appInsightsClient && eventName) {
+    const eventProperties = {
+      crn: requestBody?.crn,
+      author: req.user?.username,
+    }
+    appInsightsClient.trackEvent({ name: eventName, properties: { ...eventProperties } })
```

Comment:
No need to spread `eventProperties` here, just assign it to properties. Or even shorter, call the variable `properties` then the assignment is `... , properties }`